src/System.Security.Claims/src/System/Security/Claims/ClaimsIdentity.cs

```diff
@@ -378,13 +375,20 @@ protected virtual byte[] CustomSerializationData
         /// Allow the association of claims with this instance of <see cref="ClaimsIdentity"/>. 
         /// The claims will not be serialized or added in Clone(). They will be included in searches, finds and returned from the call to <see cref="ClaimsIdentity.Claims"/>.
         /// </summary>               
-        internal Collection<IEnumerable<Claim>> ExternalClaims
+        internal List<List<Claim>> ExternalClaims
         {
-            get { return _externalClaims; }
+            get
+            {
+                if (_externalClaims == null)
+                {
+                    _externalClaims = new List<List<Claim>>();
+                }
+                return _externalClaims;
+            }
```

Comment:
As for being thread-safe, I would keep the same semantics as netfx which is thread-safe here and Claim.Properties unless there is a precedent elsewhere for not doing so (e.g. mono or similar classes that have lazy initialization), or if it was known overkill to begin with. Using a field initializer is thread-safe, otherwise with deferred creation a lock will be needed, but some care to avoid deadlocks if sharing the same lock object.